### PR TITLE
qcontainer: slof support scsi id from 0 to 63 and lun id from 0 to 31 on vscsi adaptor

### PR DESCRIPTION
**Pull request by wangxianxian**
ID: 1837184, 1837188

Signed-off-by: Xianwang <xianwang@redhat.com>

### CODE REVIEW

**Comment by PaulYuuu** [comment]
Hello @wangxianxian , there is more than one place in qcontainer that involves the addr of spapr-vscsi bus.
https://github.com/avocado-framework/avocado-vt/blob/3fd00812165e47b7efc9ba8ad2bc1efa80ececd9/virttest/qemu_devices/qcontainer.py#L962-L968
https://github.com/avocado-framework/avocado-vt/blob/3fd00812165e47b7efc9ba8ad2bc1efa80ececd9/virttest/qemu_devices/qcontainer.py#L1794-L1797
https://github.com/avocado-framework/avocado-vt/blob/3fd00812165e47b7efc9ba8ad2bc1efa80ececd9/virttest/qemu_devices/qcontainer.py#L1828-L1829

Can you confirm that we only need to modify this one?

**Comment by wangxianxian** [comment]
@PaulYuuu thanks for your advice.
For these two slof cases, it only need to modify that one, but it's better to update other related parts, I have already updated it.

**Comment by wangxianxian** [comment]
This patch is from a product bz patch of slof:
https://patchwork.ozlabs.org/patch/1008697/
it changes the amount of IDs that we scan in SLOF from 0 to 63, to match the ID range that QEMU provides.

**Comment by PaulYuuu** [comment]
Looks good to me.
@luckyh Could you take a look? This change applies to new SLOF and QEMU, do we need to handle different version of they or just this change is ok?

**Comment by luckyh** [comment]
> Looks good to me.
> @luckyh Could you take a look? This change applies to new SLOF and QEMU, do we need to handle different version of they or just this change is ok?

@PaulYuuu checked the qemu source code and found that the range of `scsi-id` always be 0 - 63 from the ancient versions, so it's ok with this change.
```
$ git describe                             
v1.4.0-1931-g371386fb60
$ grep max_target -C3 hw/scsi/spapr_vscsi.c
static const struct SCSIBusInfo vscsi_scsi_info = {
    .tcq = true,
    .max_channel = 7, /* logical unit addressing format */
    .max_target = 63,
    .max_lun = 31,

    .transfer_data = vscsi_transfer_data,
```
@PaulYuuu @wangxianxian btw, see that the maximum number of `lun` is 31, shall we update it as well? thanks!

**Comment by wangxianxian** [comment]
@luckyh Yes, it's indeed that the range of lun id that qemu provides is 0-31, sure, I will update it.

**Comment by wangxianxian** [comment]
@PaulYuuu please help to review it again, thanks

**Comment by PaulYuuu** [comment]
@wangxianxian Please update your commit message to cover SCSI ID and LUN ID changes.

**Comment by YongxueHong** [comment]
> @wangxianxian Please update your commit message to cover SCSI ID and LUN ID changes.

Hi @wangxianxian Your commit header exceeds 69 characters, so please shorten it then give the explanation in the blew message.
e.g:
```
qcontainer: Update the scsi id and lun id address for vscsi adaptor

SLOF supports scsi id address from 0 to 63 and lun id address 
from 0 to 31 on vscsi adaptor.
```

**Comment by wangxianxian** [comment]
> > @wangxianxian Please update your commit message to cover SCSI ID and LUN ID changes.
> 
> Hi @wangxianxian Your commit header exceeds 69 characters, so please shorten it then give the explanation in the blew message.
> e.g:
> 
> ```
> qcontainer: Update the scsi id and lun id address for vscsi adaptor
> 
> SLOF supports scsi id address from 0 to 63 and lun id address 
> from 0 to 31 on vscsi adaptor.
> ```

Thanks for yhong's reviewing, I will update it later, thanks

**Comment by luckyh** [comment]
@ldoktor would you please help check it as well? thanks!

File: virttest/qemu_devices/qcontainer.py [hunk 1]
```diff
@@ -1794,7 +1794,7 @@ def define_hbas(qtype, atype, bus, unit, port, qbus, pci_bus, iothread,
                 if arch.ARCH in ('ppc64', 'ppc64le'):
                     _ = define_hbas('SCSI', 'spapr-vscsi', None, None, None,
                                     qdevices.QSCSIBus, None, iothread,
-                                    addr_spec=[8, 16384])
+                                    addr_spec=[64, 32])
```
Comment (ldoktor) [inline virttest/qemu_devices/qcontainer.py:1797]: yep, qemu claims support 0-63 and 0-31

File: virttest/qemu_devices/qcontainer.py [hunk 2]
```diff
@@ -1826,7 +1826,7 @@ def define_hbas(qtype, atype, bus, unit, port, qbus, pci_bus, iothread,
                 elif scsi_hba == 'virtio-scsi-ccw':
                     pci_bus = None
             elif scsi_hba == 'spapr-vscsi':
-                addr_spec = [8, 16384]
+                addr_spec = [64, 32]
```
Comment (ldoktor) [inline virttest/qemu_devices/qcontainer.py:1829]: Here qemu allows 0-255 and 0-16383. I'll bring this on s390x meeting to see whether it's expected

Comment (ldoktor) [reply virttest/qemu_devices/qcontainer.py:1829]: (the meeting is in ~2h and I forgot to mention I used about a month old upstream qemu)

Comment (ldoktor) [reply virttest/qemu_devices/qcontainer.py:1829]: OK so I used IRC and zkvm team would like to use the current limit, which is `[256, 16384]`. Would you please update this? (unless you have some good reason not to...)

**Pull request merged**